extensions/amp-mustache/0.2/amp-mustache.js

```diff
@@ -58,27 +63,80 @@ export class AmpMustache extends AMP.BaseTemplate {
     /** @private @const {string} */
     this.template_ = this.initTemplateString_();
 
-    mustache.parse(this.template_, /* tags */ undefined);
+    try {
+      mustache.parse(this.template_, /* tags */ this.getDelimiters_());
+    } catch (e) {
+      userAssert(false, 'Error parsing template: %s', e);
+    }
   }
 
   /**
    * @private
    * @return {string}
    */
   initTemplateString_() {
-    if (this.element.tagName == 'TEMPLATE') {
-      const content = templateContentClone(this.element);
+    const {element} = this;
+    if (element.tagName == 'TEMPLATE') {
+      const content = templateContentClone(element);
       this.processNestedTemplates_(content);
-      const container = this.element.ownerDocument.createElement('div');
+      const container = element.ownerDocument.createElement('div');
       container.appendChild(content);
       return container./*OK*/innerHTML;
-    } else if (this.element.tagName == 'SCRIPT') {
-      return this.element.textContent;
+    } else if (element.tagName == 'SCRIPT') {
+      return element.textContent;
     }
 
     return '';
   }
 
+  /**
+   * Initialize the delimiters.
+   * @return {?Array<string>} delimiters or null. Null rather than empty array
+   *     so that default mustache delimiters are used.
+   * @private
+   */
+  getDelimiters_() {
+    let delimiters = null;
+    const {element} = this;
+    if (element.hasAttribute(CUSTOM_DELIMITERS_ATTR)) {
+      const delimitersStr = element.getAttribute(CUSTOM_DELIMITERS_ATTR);
+      delimiters = delimitersStr.split(',');
+      userAssert(delimiters.length == 2,
+          'Beginning and ending delimiter is required: %s.', element);
+      userAssert(this.validDelimiters_(delimiters),
+          'Empty space and "=" are invalid delimiters');
+      if (element.tagName == 'TEMPLATE') {
+        this.encodeHtmlEntitiesInDelimiter_(delimiters);
+      }
+    }
+    return delimiters;
+  }
+
+  /**
+   * @param {Array<string>} delimiters
+   * @return {boolean}
+   * @private
+   */
+  validDelimiters_(delimiters) {
+    return delimiters.some(delimiter => {
```

Comment:
This would pass with one valid delimiter.

Reply:
ah good catch. added an additional test to test this. thanks.